packages/docusaurus-theme-classic/src/theme/DocSidebar/index.js

```diff
@@ -36,6 +36,26 @@ function DocSidebarItem({item, onItemClick, collapsible, ...props}) {
     setCollapsed((state) => !state);
   });
 
+  const activePageRelativeUrl =
+    window.location.pathname + window.location.search;
+
+  // We need to know if the category item
+  // is the parent of the active page
+  // If it is, this returns true and make sure to highlight this category
+  const isCategoryOfActivePage = () => {
+    // Make sure we have items
+    if (typeof items !== 'undefined') {
+      return items.some((categoryItem) => {
+        // Grab the category item's href
+        const childHref = categoryItem.href;
+        // Compare it to the current active page
+        return activePageRelativeUrl === childHref;
+      });
+    }
+
+    return false;
+  };
+
```

Comment:
@lex111 I added a fix for the category highlighting.

Here's a demo:
![2020-04-29 16 11 34](https://user-images.githubusercontent.com/3806031/80655710-35934d80-8a34-11ea-9c8d-24d4c615f307.gif)

Please let me know if you have any feedback/thoughts.